src/execution_plan/execution_plan_build/build_projection_ops.c

```diff
@@ -203,8 +237,20 @@ static inline void _buildProjectionOps(ExecutionPlan *plan,
 	if(order_clause) {
 		AST_PrepareSortOp(order_clause, &sort_directions);
 		order_exps = _BuildOrderExpressions(projections, order_clause);
-		// Merge order expressions into the projections array.
-		_combine_projection_arrays(&projections, order_exps);
+		is_order_exps_use_projected_aliases = _is_order_exps_use_projected_aliases(order_exps, projected_aliases);
+		if(!is_order_exps_use_projected_aliases) {
+			// Merge order expressions into the projections array.
+			_combine_projection_arrays(&projections, order_exps);
+		} else {
+			// Add another projection because order by uses expression from current record
+			array_clone_with_cb(order_projections, order_exps, AR_EXP_Clone);
```

Comment:
This logic doesn't quite work because a cloned function call will not necessarily produce the same result as the original function. For example, in the failing test:
```
MATCH (a)-[e:E]-(b) RETURN ID(e) AS id, a.val, b.val ORDER BY id, a.val, b.val
```
The first projection correctly resolves `a.val` and `b.val`, but the new project op will return NULL due to failed property lookups on inaccessible entities.